sdk/keyvault/keyvault-certificates/src/generated/api/operations.ts

```diff
@@ -0,0 +1,2698 @@
+ * The GetCertificates operation returns the set of certificates resources in the
+ * specified key vault. This operation requires the certificates/list permission.
+ */
+export function getCertificates(
+  context: Client,
+  options: GetCertificatesOptionalParams = { requestOptions: {} }
+): PagedAsyncIterableIterator<CertificateItem> {
+  return buildPagedAsyncIterator(
+    context,
+    () => _getCertificatesSend(context, options),
+    _getCertificatesDeserialize,
+    { itemName: "value", nextLinkName: "nextLink" }
+  );
+}
+
+export function _deleteCertificateSend(
+  context: Client,
+  certificateName: string,
+  options: DeleteCertificateOptionalParams = { requestOptions: {} }
+): StreamableMethod<DeleteCertificate200Response | DeleteCertificateDefaultResponse> {
+  return context
+    .path("/certificates/{certificateName}", certificateName)
+    .delete({ ...operationOptionsToRequestParameters(options) });
+}
+
+export async function _deleteCertificateDeserialize(
+  result: DeleteCertificate200Response | DeleteCertificateDefaultResponse
+): Promise<DeletedCertificateBundle> {
+  if (isUnexpected(result)) {
+    throw createRestError(result);
+  }
+
+  return {
+    id: result.body["id"],
+    kid: result.body["kid"],
+    sid: result.body["sid"],
+    x509Thumbprint:
+      typeof result.body["x5t"] === "string"
+        ? stringToUint8Array(result.body["x5t"], "base64")
+        : result.body["x5t"],
+    policy: !result.body.policy
+      ? undefined
+      : {
+          id: result.body.policy?.["id"],
+          keyProperties: !result.body.policy?.key_props
+            ? undefined
+            : {
+                exportable: result.body.policy?.key_props?.["exportable"],
+                keyType: result.body.policy?.key_props?.["kty"] as JsonWebKeyType,
+                keySize: result.body.policy?.key_props?.["key_size"],
+                reuseKey: result.body.policy?.key_props?.["reuse_key"],
+                curve: result.body.policy?.key_props?.["crv"] as JsonWebKeyCurveName,
+              },
+          secretProperties: !result.body.policy?.secret_props
+            ? undefined
+            : {
+                contentType: result.body.policy?.secret_props?.["contentType"],
+              },
+          x509CertificateProperties: !result.body.policy?.x509_props
+            ? undefined
+            : {
+                subject: result.body.policy?.x509_props?.["subject"],
+                ekus: result.body.policy?.x509_props?.["ekus"],
+                subjectAlternativeNames: !result.body.policy?.x509_props?.sans
+                  ? undefined
+                  : {
+                      emails: result.body.policy?.x509_props?.sans?.["emails"],
+                      dnsNames: result.body.policy?.x509_props?.sans?.["dns_names"],
+                      upns: result.body.policy?.x509_props?.sans?.["upns"],
+                    },
+                keyUsage: result.body.policy?.x509_props?.["key_usage"] as KeyUsageType[],
```

Comment:
Had to add this cast `as KeyUsageType[]` to fix the generated code not compiling. This is probably an emitter bug related to the recent extensible enum work.